### PR TITLE
[MUI-413] Create Table Component

### PR DESCRIPTION
**Pull request by p1geon**
# Description

Create Table Component

## Jira ticket(optional)

[MUI-413](https://minterestfinance.atlassian.net/browse/MUI-413)

## Type of change

Please pick options that are relevant.

- [ ] Bug fix OR refactoring (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature or refactoring that would cause existing functionality to not work as expected)

image

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -19,7 +19,9 @@
   "dependencies": {
     "@emotion/react": "^11.9.3",
     "@emotion/styled": "^11.9.3",
-    "@mui/material": "5.8.7",
+    "@mui/icons-material": "^5.8.4",
```
Comment (gavrilin-remote) [inline package.json:22]: remove icons dependecies

File: src/components/Table/EnchancedTable.tsx [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+  return (
+    <>
+      {addHeader && (
+        <div
+          style={{
+            paddingTop: headerPaddingTop || 0,
+            paddingBottom: headerPaddingBottom || 0,
+            paddingLeft: headerPaddingLeft || 0,
+            border: '1px solid rgba(12, 45, 156, 0.08)',
```
Comment (gavrilin-remote) [inline src/components/Table/EnchancedTable.tsx:55]: We can remove title and borders from table. Leave only datagrid

File: src/components/Table/EnchancedTable.tsx [hunk 3]
```diff
@@ -0,0 +1,126 @@
+import * as React from 'react';
+
+import { SxProps, Theme } from '@mui/material';
+import { DataGrid, GridColDef, GridCellParams } from '@mui/x-data-grid';
+
+import { Typography } from 'components';
+
+// TODO fix row type
+type EnchancedTableProps = {
```
Comment (gavrilin-remote) [inline src/components/Table/EnchancedTable.tsx:9]: replace with interface EnchancedTableProps extends DataGridProps {
...
}

and remove overlap props from EnchancedTableProps

File: package.json (hunk 1 above)
Comment (p1geon) [reply package.json:22]: removed

File: src/components/Table/EnchancedTable.tsx (hunk 2 above)
Comment (p1geon) [reply src/components/Table/EnchancedTable.tsx:55]: title is already passing as a prop (it is called headerTitle though), are you talking about something else?

removed border from there and now passing it as headerStyels obj

File: src/components/Table/EnchancedTable.tsx (hunk 3 above)
Comment (p1geon) [reply src/components/Table/EnchancedTable.tsx:9]: Fixed. I think) Please doublecheck

**Review by mr-teddy-bear — APPROVED** [review]

File: src/components/Table/EnchancedTable.tsx [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        },
+        '.MuiDataGrid-columnHeader:focus': {
+          outline: 'none',
+        },
+
+        '.MuiDataGrid-row': {
+          transition: 'transform 0.3s ease',
+          // 'z-index': 1,
+        },
+        // '.MuiDataGrid-virtualScroller': {
```
[the remark this answers is not in the document]
Comment (p1geon) [reply src/components/Table/EnchancedTable.tsx]: removed

File: src/components/Table/EnchancedTable.tsx [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+          },
+          '.MuiDataGrid-columnHeader:focus': {
+            outline: 'none',
+          },
+
+          '.MuiDataGrid-row': {
+            transition: 'transform 0.3s ease',
+            // 'z-index': 1,
+          },
+          // '.MuiDataGrid-virtualScroller': {
```
[the remark this answers is not in the document]
Comment (p1geon) [reply src/components/Table/EnchancedTable.tsx]: removed

**Review by gavrilin-remote — APPROVED** [review]

**Pull request merged**